code/datums/components/manual_heart.dm

```diff
@@ -0,0 +1,170 @@
+/**
+ * Manual heart pumping component. Requires the holder to pump their heart manually every
+ * so often or die.
+ *
+ * Mainly used by the cursed heart.
+ */
+/datum/component/manual_heart
+	/// The action for pumping your heart
+	var/datum/action/cooldown/manual_heart/pump_action
+	var/last_pump = 0
+	var/add_colour = TRUE //So we're not constantly recreating colour datums
```

Comment:
```suggestion
	/// If true, add a screen tint on the next process
	var/add_colour = TRUE 
```
Please document each var, even if the name is self explanatory. The comment attached to it actually isn't very helpful if we don't know what the var does at this point 😅